pkg/webui/console/containers/devices-table/index.js

```diff
@@ -132,14 +139,19 @@ class DevicesTable extends React.Component {
   constructor(props) {
     super(props)
 
-    this.getDevicesList = filters => getDevicesList(props.appId, filters, ['name'])
+    this.getDevicesList = filters =>
+      getDevicesList(props.appId, filters, ['name'], { withLastSeen: true })
   }
 
   @bind
   baseDataSelector(state) {
-    const { mayCreateDevices } = this.props
+    const { mayCreateDevices, appId } = this.props
+    const devices = selectDevices(state)
+    for (const device of devices) {
+      device._derivedLastSeen = selectDeviceDerivedLastSeen(state, appId, device.ids.device_id)
```

Comment:
This updates device objects in the redux store. The only place to do that is in reducers. Is there any way to set `derivedLastSeen` before entities are selected for the end device table?

Reply:
Indeed, didn't mean to do that. I now just clone the device object before modifying it.